server/api/crud/model_monitoring/deployment.py

```diff
@@ -177,41 +196,90 @@ def deploy_model_monitoring_controller(
+        :param controller_image:            The image of the model monitoring controller function.
+                                            By default, the image is mlrun/mlrun.
+        :param overwrite:                   If true, overwrite the existing model monitoring controller.
+                                            By default, False.
+
 
         :return: Model monitoring controller job as a runtime function.
         """
 
-        self.deploy_model_monitoring_writer_application(
+        if not overwrite:
+            logger.info(
+                "Checking if model monitoring controller is already deployed",
+                project=project,
+            )
+            try:
+                # validate that the model monitoring stream has not yet been deployed
+                mlrun.runtimes.nuclio.function.get_nuclio_deploy_status(
+                    name=mm_constants.MonitoringFunctionNames.APPLICATION_CONTROLLER,
+                    project=project,
+                    tag="",
+                    auth_info=auth_info,
+                )
+                logger.info(
+                    "Detected model monitoring controller processing function already deployed",
+                    project=project,
+                )
+                return
+            except mlrun.errors.MLRunNotFoundError:
+                pass
+        logger.info(
+            "Deploying model monitoring controller processing function",
             project=project,
-            model_monitoring_access_key=model_monitoring_access_key,
-            db_session=db_session,
-            auth_info=auth_info,
-            tracking_policy=tracking_policy,
         )
 
-        return self.deploy_model_monitoring_batch_processing(
+        fn = self._get_model_monitoring_batch_function(
             project=project,
             model_monitoring_access_key=model_monitoring_access_key,
             db_session=db_session,
             auth_info=auth_info,
-            tracking_policy=tracking_policy,
+            image=controller_image,
             function_name=mm_constants.MonitoringFunctionNames.APPLICATION_CONTROLLER,
         )
+        self._apply_access_key_and_mount_function(
+            project,
+            fn,
+            model_monitoring_access_key,
+            auth_info,
+            mm_constants.MonitoringFunctionNames.APPLICATION_CONTROLLER,
+        )
+        minutes = base_period
+        hours = days = 0
+        batch_dict = {
+            mm_constants.EventFieldType.MINUTES: minutes,
+            mm_constants.EventFieldType.HOURS: hours,
+            mm_constants.EventFieldType.DAYS: days,
+        }
+        fn.set_env(
+            mm_constants.EventFieldType.BATCH_INTERVALS_DICT,
+            json.dumps(batch_dict),
+        )
+
+        fn.add_trigger(
+            "cron_interval",
+            spec=nuclio.CronTrigger(interval=f"{base_period}m"),
+        )
```

Comment:
Looks neat!
To make sure I understand, trigger events will be sent according to the set interval, but the function is always up - a daemon. Right?

Reply:
Yep :)